services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/component/DigitalGreenCertificateStructureProvider.java

```diff
@@ -98,13 +103,16 @@ private DirectoryOnDisk constructArchiveToPublish(DigitalGreenCertificate dgcCon
       onboardedCountries
           .addWritable(new FileOnDisk("export.bin", dgcToCborMapping.constructCborCountries()));
       logger.info("Onboarded countries archive has been added to the DGC distribution folder");
+
+      return Optional.of(new DistributionArchiveSigningDecorator(onboardedCountries, cryptoProvider,
```

Comment:
I see that `getOnboardedCountriesArchive().ifPresent(dgcDirectory::addWritable);` looks nice, but is there actually a benefit of wrapping this `Optional.of(new DistributionArchiveSigningDecorator(onboardedCountries, cryptoProvider,` in an Optional? The point of failure is `dgcToCborMapping.constructCborCountries()`, so I dont see the benefit here (codingwise) and I would maybe use a Consumer....Maybe we can discuss this :)